### PR TITLE
Cassandra gocql v2 support

### LINKED ISSUE #45: Adapt `gocql-driver-matrix` to support `cassandra-gocql-driver/v2`
It uses a different package name `github.com/apache/cassandra-gocql-driver` thus script is unable to process test results
```
17:21:40  PASS
17:21:40  ok  	github.com/apache/cassandra-gocql-driver/v2	1.008s
17:21:40  INFO:cluster:Removing test cluster...
17:21:41  INFO:cluster:test cluster removed
17:21:41  Warning: Could not find testsuite with name 'github.com/gocql/gocql' in /gocql-driver-matrix/xunit/v2.0.0/xunit.upstream.v3.v2.0.0.xml_part_0
17:21:41  Warning: Could not find testsuite with name 'github.com/gocql/gocql' in /gocql-driver-matrix/xunit/v2.0.0/xunit.upstream.v3.v2.0.0.xml_part_1
```

### PR DESCRIPTION
**Pull request by nikagra**
Fixes #45 

## Overview
This pull request adds support for the new Cassandra gocql driver v2.0.0 while maintaining backward compatibility with existing Scylla and upstream drivers.

### Problem
Cassandra gocql v2.0.0 uses a different module name (`github.com/apache/cassandra-gocql-driver/v2`) than the traditional `github.com/gocql/gocql`

### Solution
Added `get_driver_module()` function to dynamically extract module names from `go.mod` files

### Implementation 
Module detection happens after git checkout/patching to ensure version-specific module names are used

## Testing
* [Verified](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/mhradovich/job/gocql-driver-matrix/4/) with both traditional gocql drivers and new Cassandra v2.0.0 driver
* Handles edge cases like missing go.mod files and failed test executions

### CODE REVIEW

File: processjunit.py [hunk 1]
```diff
@@ -112,8 +112,8 @@ def save_after_analysis(self, driver_version: str, protocol: int, gocql_driver_t
         :param protocol: The cqlsh native protocol number
         :param gocql_driver_type: The driver type - can be "scylla" or "upstream"
```
[the remark this answers is not in the document]
Comment (dkropachev) [reply processjunit.py]: +1

File: run.py [hunk 2]
```diff
@@ -171,6 +172,28 @@ def run(self) -> ProcessJUnit:
...
+    """
+    DEFAULT_GOCQL_MODULE = "github.com/gocql/gocql"
+    go_mod_file = os.path.join(gocql_driver_git, "go.mod")
+    if not os.path.isfile(go_mod_file):
+        logging.error(f"go.mod file not found in {gocql_driver_git}, defaulting module name to '{DEFAULT_GOCQL_MODULE}'")
+        return DEFAULT_GOCQL_MODULE
+    with open(go_mod_file, "r") as f:
+        for line in f:
+            if line.startswith("module "):
+                module = line.split("module ")[1].strip()
```
[the remark this answers is not in the document]
Comment (dkropachev) [reply run.py]: i would line.split()[1].strip()

File: scripts/entrypoint.sh [hunk 3]
```diff
@@ -11,5 +11,6 @@ if [ ! -d "/gocql-driver-matrix" ]
   then echo "Error. /gocql-driver-matrix is not mounted. Mount gocql-driver-matrix repo to /gocql-driver-matrix in docker run command."
   exit 1
 fi
+pip install --upgrade pip --quiet
```
[the remark this answers is not in the document]
Comment (nikagra) [reply scripts/entrypoint.sh]: We already have it in Docker file, but this state is not maintained due to `--tmpfs ${HOME}/.cache \` Docker param in `run_test.sh`:
```
COPY requirements.txt .
RUN pip install --no-cache-dir --upgrade pip
RUN pip install --no-cache-dir -r requirements.txt
```
